pymc/tests/test_distributions.py

```diff
@@ -2216,20 +2216,45 @@ def test_multinomial(self, n):
             lambda value, n, p: scipy.stats.multinomial.logpmf(value, n, p),
         )
 
-    def test_multinomial_invalid(self):
-        # Test non-scalar invalid parameters/values
+    def test_multinomial_invalid_value(self):
+        # Test passing non-scalar invalid parameters/values to an otherwise valid Multinomial, 
+        # evaluates to -inf 
         value = np.array([[1, 2, 2], [4, 0, 1]])
-
-        invalid_dist = Multinomial.dist(n=5, p=[-1, 1, 1], size=2)
-        # TODO: Multinomial normalizes p, so it is impossible to trigger p checks
-        # with pytest.raises(ParameterValueError):
-        with does_not_raise():
-            pm.logp(invalid_dist, value).eval()
-
         value[1] -= 1
         valid_dist = Multinomial.dist(n=5, p=np.ones(3) / 3)
         assert np.all(np.isfinite(pm.logp(valid_dist, value).eval()) == np.array([True, False]))
-
+    
+    def test_multinomial_negativep_value_error(self):
+        # test passing a list/numpy with negative p raises an immediate error
+        with pytest.raises(ValueError):
```

Comment:
You can add a `match` string with the expected error message in `pytest.raises` for a more strict test.

Reply:
Added `match` string to tests expecting UserWarning and ValueError.